ceph_iscsi_config/utils.py

```diff
@@ -86,7 +86,11 @@ def resolve_ip_addresses(addr):
         infos = socket.getaddrinfo(addr, 0)
         for info in infos:
             if info[0] in families:
-                addrs.add(info[4][0])
+                # remove interface scope suffix and IPv4-over-IPv6 prefix
+                addr = info[4][0]
+                addr = addr.rsplit('%', 1)[0]
```

Comment:
Nit: should we just skip over any link-local addresses? i.e. if it contains `%`, just go to the next one?

Reply:
The bug/issue here was that requires_restricted_auth gets the link local address sometimes (I have still have only been able to replicate it by adding a mis-set address, and cannot figure out why other users see it) in the request.remote_addr value. That function would chop off the zone/interface info after the %, so above I chopped it to match what the caller was expecting. If I do not chop the interface and return the value then we will fail to match in requires_restricted_auth:

```
        if remote_addr not in gw_ips:
            return jsonify(message="API access not available to "
                                   "{}".format(remote_addr)), 403
```

An alt fix would be to just remove the line in requires_restricted_auth:

`remote_addr = request.remote_addr.rsplit('%', 1)[0]
`
but I was thinking there might be future resolve_ip_addreses users that want the address. I am really not sure though.

I do not know ipv6 well, but I noticed you added the requires_restricted_auth code. The only time I have seen the zone info is when it has the interface info for link local addresses, but it sounds like you can get the zone info for non link local addresses.

Did you by any chance see the interface info after the "%" and add that code in requires_restricted_auth for the for link local case?

Reply:
I've only seen scopes on link-local addresses -- and those are added to each interface where IPv6 is enabled (even if the same interface has a real IPv6 address). I'm fine dropping support for link-local (i.e. strip them out everywhere) if that's doable.